drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/panel/AnalysisReportScreen.java

```diff
@@ -153,4 +164,33 @@ public void showStatus( final Status status ) {
                               status.getEnd(),
                               status.getTotalCheckCount() );
     }
+
+    private Predicate<Issue> shouldNotBeMergedInReport = new Predicate<Issue>() {
+        @Override
+        public boolean test(Issue issue) {
+            return !( issue.getExplanationType() == ExplanationType.EMPTY_RULE ||
+                      issue.getExplanationType() == ExplanationType.MISSING_RESTRICTION ||
+                      issue.getExplanationType() == ExplanationType.MISSING_ACTION );
+        }
+    };
+
+    private void mergeIssues( Set<Issue> targetIssueSet, List<Issue> issues, ExplanationType type) {
```

Comment:
Here's how I would implement the changes in this class. But feel free to ignore this :-)

```
    // Issues that we want to report on single line (the sets of lines where they occurred will be merged)
    private Set<ExplanationType> MERGEABLE_ISSUES = EnumSet.of(
            ExplanationType.EMPTY_RULE,
            ExplanationType.MISSING_RESTRICTION,
            ExplanationType.MISSING_ACTION
    );

    private ArrayList<Issue> getIssues(final AnalysisReport report) {
        Set<Issue> mergedIssues = MERGEABLE_ISSUES.stream()
                .map(typeToMerge -> mergeIssues(report.getAnalysisData(), typeToMerge))
                .filter(Optional::isPresent).map(Optional::get).collect(toSet());

        Set<Issue> individualIssues = report.getAnalysisData().stream()
                .filter(issue -> !MERGEABLE_ISSUES.contains(issue.getExplanationType()))
                .collect(toSet());

        final IssuesSet allIssues = new IssuesSet();
        allIssues.addAll(mergedIssues);
        allIssues.addAll(individualIssues);

        return new ArrayList<>(allIssues);
    }

    private Optional<Issue> mergeIssues(List<Issue> issues, ExplanationType typeToMerge) {
        Set<Issue> issuesToMerge = issues.stream()
                .filter(issue -> issue.getExplanationType() == typeToMerge)
                .collect(toSet());

        Set<Integer> affectedRows = issuesToMerge.stream()
                .flatMap(issue -> issue.getRowNumbers().stream())
                .collect(toSet());

        return issuesToMerge.stream()
                .findFirst() // Will be Optional.empty() if no issue of "typeToMerge" was present
                .map(issue -> new Issue(issue.getSeverity(), typeToMerge, affectedRows));
    }
```